### DISCUSSION TITLE
What is the refetch() type for child component props

### DISCUSSION THREAD

**Discussion by Re4GD** [trpc/trpc#3611] [Q&A]
I have a useQuery in my parent component. I want to give this useQuery's refetch function to a child component as a prop, and use `await refetch()` in the child. 

Looking at the documentation, I have found where the refetch is defined as a part of an interface:

```
export interface QueryObserverBaseResult<TData = unknown, TError = unknown> {
    data: TData | undefined;
    ...
    refetch: <TPageData>(options?: RefetchOptions & RefetchQueryFilters<TPageData>) => Promise<QueryObserverResult<TData, TError>>;
    ...
}
```

An example of inferring router inputs and outputs
```
export type RouterInputs = inferRouterInputs<AppRouter>;
export type RouterOutputs = inferRouterOutputs<AppRouter>;
```

But I have not found a trpc way of inferring the type of refetch, or even the type of useQuery (I think if I found the type of useQuery, I could have reached a result). How do I infer the type of the refetch function to pass onto the child component's FC<> interface?

**Comment by Lok3sh-kumar** [comment]
Hey! @Re4GD this is how i do it!
```
import type { QueryObserverResult } from "@tanstack/react-query";

interface ChildComponentProps {
  yourRefetchProp: QueryObserverResult["refetch"];
}
```

Hope this works for you! Cheers!!
